src/campaign/entity/Campaign.entity.ts

```diff
@@ -0,0 +1,85 @@
+import {
+    Entity,
+    PrimaryGeneratedColumn,
+    Column,
+    ManyToOne,
+    JoinColumn,
+} from 'typeorm';
+import { CampaignStatus } from '../status/campaign-status.enum';
+import { MailContentType } from 'src/template/types/Mail-content.type';
+import { Template } from 'src/template/entity/Template.entity';
+import { EmailProvider } from '../../email-provider/entity/Email-provider.entity';
+
+@Entity()
+export class Campaign {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column()
+    name: string;
+
+    @Column()
+    description: string;
+
+    @Column({
+        type: 'enum',
+        enum: CampaignStatus,
+        default: CampaignStatus.PREPARING,
+    })
+    status: CampaignStatus;
+
+    @ManyToOne(() => Template, (template) => template.campaigns, {
+        eager: true,
+    })
+    @JoinColumn()
+    template: Template;
+
+    @ManyToOne(
+        () => EmailProvider,
+        (eMailProvider) => eMailProvider.campaigns,
+        {
+            eager: true,
+        },
+    )
+    @JoinColumn()
+    eMailProvider: EmailProvider;
+
+    @Column('json', { nullable: true })
+    content: MailContentType;
+
+    @Column({ nullable: true })
+    eMailTitle: string;
+
+    @Column('simple-array', { nullable: true })
+    eMails: string[];
+
+    @Column()
+    manager: string; // user id from Auth0
+
+    @Column('simple-array', { nullable: true })
+    employees: string[];
+
+    @Column()
+    updatedAt: Date;
+
+    @Column()
+    updatedBy: string; // user id from Auth0
+
+    @Column({ nullable: true })
+    preparedAt: Date;
+
+    @Column({ nullable: true })
+    preparedBy: string; // user id from Auth0
+
+    @Column({ nullable: true })
+    executedAt: Date;
+
+    @Column({ nullable: true })
+    executedBy: string; // user id from Auth0
+
+    @Column({ nullable: true })
+    errorAt: Date;
```

Comment:
nie do konca jestem fanem takiego podjesica - jezeli masz powiedzmy blad w aplikacji i twoja aplikacja nie bedzie mogla nawiazac polaczenia z baza danych to tracisz te dane, takie cos powinno byc wiadome z logow w twojej aplikacji, zamiast takich 2 kolumn error* dalbym po prostu status czy sie wyslalo czy sie nie wyslalo i jakiekolwiek przyczyny tego szukalbym w logach

Chyba ze macie konkretnie jakis pomysl na to to slucham bo jestem ciekawy

Reply:
ok jasne. To była luźna propozycja, wywalę te 2 kolumny.